### PR TITLE
rpc: quantize db size up to 5 GB in restricted mode

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: src/rpc/core_rpc_server.cpp [hunk 1]
```diff
@@ -214,7 +220,9 @@ namespace cryptonote
       boost::shared_lock<boost::shared_mutex> lock(m_bootstrap_daemon_mutex);
       res.was_bootstrap_ever_used = m_was_bootstrap_ever_used;
     }
-    res.database_size = restricted ? 0 : m_core.get_blockchain_storage().get_db().get_database_size();
+    res.database_size = m_core.get_blockchain_storage().get_db().get_database_size();
+    if (restricted)
+      res.database_size = round_up(res.database_size, 5 * 1024 * 1024 * 1024);
```
Comment (xiphon) [inline src/rpc/core_rpc_server.cpp]: integer overflow

```suggestion
      res.database_size = round_up(res.database_size, 5ull * 1024 * 1024 * 1024);
```

File: src/rpc/core_rpc_server.cpp [hunk 2]
```diff
@@ -65,6 +65,12 @@ namespace
       reasons += ", ";
     reasons += reason;
   }
+
+  // assumes power of 2 quantum
+  uint64_t round_up(uint64_t value, uint64_t quantum)
+  {
+    return (value + quantum - 1) & ~(quantum - 1);
```
Comment (xiphon) [inline src/rpc/core_rpc_server.cpp]: Won't work. Should be
```suggestion
    return (value + quantum - 1) / quantum * quantum;
```

Comment (moneromooo-monero) [reply src/rpc/core_rpc_server.cpp]: Why would it not work ? I'll change to that anyway.

Comment (xiphon) [reply src/rpc/core_rpc_server.cpp]: due to the const we are using, it doesn't meet `power of 2` requirement

**Comment by vtnerd** [comment]
Is this for privacy reasons?

**Comment by moneromooo-monero** [comment]
Yes.

**Pull request merged**